x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/model/rule_assets/prebuilt_rule_asset.ts

```diff
@@ -6,13 +6,70 @@
  */
 
 import * as z from 'zod';
+import type {
+  EqlRuleCreateFields,
+  QueryRuleCreateFields,
+  SavedQueryRuleCreateFields,
+  ThresholdRuleCreateFields,
+  ThreatMatchRuleCreateFields,
+  MachineLearningRuleCreateFields,
+  NewTermsRuleCreateFields,
+  EsqlRuleCreateFields,
+} from '../../../../../../common/api/detection_engine/model/rule_schema';
 import {
   RuleSignatureId,
   RuleVersion,
   BaseCreateProps,
   TypeSpecificCreateProps,
 } from '../../../../../../common/api/detection_engine/model/rule_schema';
 
+type BaseFields = keyof BaseCreateProps;
+type SpecificFields =
+  | keyof Omit<EqlRuleCreateFields, 'type'>
+  | keyof Omit<QueryRuleCreateFields, 'type'>
+  | keyof Omit<SavedQueryRuleCreateFields, 'type'>
+  | keyof Omit<ThresholdRuleCreateFields, 'type'>
+  | keyof Omit<ThreatMatchRuleCreateFields, 'type'>
+  | keyof Omit<MachineLearningRuleCreateFields, 'type'>
+  | keyof Omit<NewTermsRuleCreateFields, 'type'>
+  | keyof Omit<EsqlRuleCreateFields, 'type'>;
+
+type AllRuleProperties = BaseFields | SpecificFields;
+
+/**
+ * The PrebuiltRuleAsset is created out of the Rule schema types defined in our OpenAPI schemas.
+ * However, we don't need all the fields in the rule schemas to be present in the PrebuiltRuleAsset.
+ * We manually exclude those unwanted fields here.
+ *
+ * Reference ticket:
+ * https://github.com/elastic/kibana/issues/180393
+ *
+ */
+type OmittedProperties = Extract<
+  AllRuleProperties,
+  | 'actions'
+  | 'throttle'
+  | 'meta'
+  | 'output_index'
+  | 'namespace'
+  | 'alias_purpose'
+  | 'alias_target_id'
+  | 'outcome'
+>;
+
+const PROPERTIES_TO_OMIT: Record<OmittedProperties, true> = {
+  actions: true,
+  throttle: true,
+  meta: true,
+  output_index: true,
+  namespace: true,
+  alias_purpose: true,
+  alias_target_id: true,
+  outcome: true,
+};
+
+const RuleAssetBaseCreateProps = BaseCreateProps.omit(PROPERTIES_TO_OMIT);
```

Comment:
This can be simplified without loosing type safety to:

```suggestion
/**
 * The PrebuiltRuleAsset schema is created based on the rule schema defined in our OpenAPI specs.
 * However, we don't need all the rule schema fields to be present in the PrebuiltRuleAsset.
 * We omit some of them because they are not present in https://github.com/elastic/detection-rules.
 * Context: https://github.com/elastic/kibana/issues/180393
 */
const PROPS_REMOVED_FROM_PREBUILT_RULE_ASSET: Partial<Record<keyof BaseCreateProps, true>> = {
  actions: true,
  throttle: true,
  meta: true,
  output_index: true,
  namespace: true,
  alias_purpose: true,
  alias_target_id: true,
  outcome: true,
};
```

Reply:
Updated the suggestion above. With the previous one the `BaseCreateProps.omit(PROPS_REMOVED_FROM_PREBUILT_RULE_ASSET)` schema was too loose and incompatible with the diffing logic.

Reply:
Addressed in https://github.com/elastic/kibana/pull/186615/commits/979cbe7a99f6c48e4fd058f2d6cd87f009bdde43